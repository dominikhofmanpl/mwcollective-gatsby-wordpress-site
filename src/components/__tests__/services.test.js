import * as React from 'react'
import renderer from "react-test-renderer"
import { StaticQuery } from 'gatsby'
import iconService from '../../assets/services-icon.svg'
import ServiceItem from '../startpage/serviceitem'
import ReactHtmlParser, {  } from 'react-html-parser'

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
        pageBy : {
            services_box : {
                servicesBox : {
                    iconService : {
                        sourceUrl
                    },
                    titleService,
                    contentService
                }
            }        
        }
    })
  )
})
describe("Service", () => {
  it("renders correctly", () => {
    const data = {
        pageBy : {
            services_box : {
                servicesBox : {
                    iconService : {
                        sourceUrl: `${iconService}`,
                    },
                    titleService: `WnÄ™trza`,
                    contentService: `Nunc porta diam ex, vel laoreet nisl placerat sit amet. Praesent pulvinar auctor nisi, sodales consequat arcu tincidunt id. Phasellus condimentum nunc in lectus lacinia rutrum id eu lorem.`
                }
            }        
        }
    }
    const tree = renderer.create(<ServiceItem
            serviceIcon={data.pageBy.services_box.servicesBox.iconService}
            serviceTitle={data.pageBy.services_box.servicesBox.titleService}
            serviceDescription={ReactHtmlParser(data.pageBy.services_box.servicesBox.contentService)}
        ></ServiceItem>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})