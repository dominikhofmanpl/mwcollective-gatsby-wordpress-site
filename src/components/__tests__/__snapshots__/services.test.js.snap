// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service renders correctly 1`] = `
<div
  className="flex flex-col py-4 md:mr-4"
>
  <img
    alt=""
    className="my-3"
    src={
      Object {
        "sourceUrl": "test-file-stub",
      }
    }
    srcset=""
    style={
      Object {
        "width": "4rem",
      }
    }
  />
  <h3
    className="mont-semi my-1"
  >
    WnÄ™trza
  </h3>
  <p
    className="inriasans-reg my-3"
  >
    Nunc porta diam ex, vel laoreet nisl placerat sit amet. Praesent pulvinar auctor nisi, sodales consequat arcu tincidunt id. Phasellus condimentum nunc in lectus lacinia rutrum id eu lorem.
  </p>
</div>
`;
